props/blaster.h

```diff
@@ -77,10 +83,53 @@ class Blaster : public PropBase {
       } else {
         hybrid_font.PlayCommon(&SFX_blast);
       }
+
+      shots_fired_++;
 #endif
     }
   }
 
+  virtual void Reload() {
+    shots_fired_ = 0;
+
+#ifdef ENABLE_AUDIO
+    if (SFX_reload.files_found()) {
+      hybrid_font.PlayCommon(&SFX_reload);
+    }
+#endif
+  }
+
+  virtual void ClipOut() {
+    if (max_shots_ != -1) shots_fired_ = max_shots_;
+
+#ifdef ENABLE_AUDIO
+    if (SFX_clipout.files_found()) {
+      hybrid_font.PlayCommon(&SFX_clipout);
+    }
+#endif
+  }
+
+  virtual void ClipIn() {
+#ifdef ENABLE_AUDIO
+    if (SFX_clipin.files_found()) {
+      hybrid_font.PlayCommon(&SFX_clipin);
+    }
+#endif
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  virtual void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+
+    if (!SaberBase::IsOn()) {
+      On();
+    }
+
+#ifdef ENABLE_AUDIO
+    hybrid_font.SB_On();
```

Comment:
Already handled by On().

Reply:
Still think it's called by On():

https://github.com/profezzorn/ProffieOS/blob/4c2f05f90ddd057ee38d7a500c741b4ebd666a8a/common/saber_base.h#L49
https://github.com/profezzorn/ProffieOS/blob/4c2f05f90ddd057ee38d7a500c741b4ebd666a8a/props/prop_base.h#L88

Also SB_On() should only be called by SaberBase::DoOn();